Switch from scratch to empty (#1068)
opencontainers/image-spec: manifest.md

}
```

## Guidance for a filler descriptor

*Implementers note*: The following is considered GUIDANCE for portability.

Parts of the spec necessitate including a descriptor to a blob where some implementations of artifacts do not have associated content.
While an empty blob (`size` of 0) may be preferable, practice has shown that not to be ubiquitously supported.
The media type `application/vnd.oci.filler.v1+json` (`MediaTypeFiller`) has been specified for a descriptor that has no content for the implementation.

> The blob payload is the most minimal content that is still valid JSON object: `{}` (`size` of 2).

Reviewer: ```suggestion
The blob payload is the most minimal content that is still a valid JSON object: `{}` (`size` of 2).
```

> The blob payload is the most minimal content that is still valid JSON object: `{}` (`size` of 2).

Reviewer: ```suggestion
The blob payload is the most minimal content that is still a valid JSON object: `{}` (`size` of 2, [base64](descriptor.md#embedded-content) of `e30=`).
```

Something like this? :eyes: (ref https://github.com/opencontainers/image-spec/pull/1068/files#r1207421094)

Author: I added a line indicating it was optional to minimize the risk someone will get confused.

The blob digest of `{}` is `sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a`.

The resulting descriptor shown here is also defined in reference code as `FillerDescriptor`:

```json,title=filler%20config&mediatype=application/vnd.oci.descriptor.v1%2Bjson
{
  "mediaType": "application/vnd.oci.filler.v1+json",
  "size": 2,

>   "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a"

Reviewer: It might be a little "extra", but this seems like a nice place IMO for us to encourage use of the `data` field (#826) too: :innocent:

```suggestion
  "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
  "data": "e30="
```

Author: I started to do that, and worried that people might not realize that it is base64 encoded `{}`. That can probably be mitigated with another line in the description above.

}
```

## Guidelines for Artifact Usage

Content other than OCI container images MAY be packaged using the image manifest.
When this is done, the `config.mediaType` value MUST be set to a value specific to the artifact type or the [filler value](#guidance-for-a-filler-descriptor).
If the `config.mediaType` is set to the filler value, the `artifactType` MUST be defined.
If the artifact does not need layers, a single layer SHOULD be included with a non-zero size.

> The suggested content for an unused layer is the [filler descriptor](#guidance-for-a-filler-descriptor).

Reviewer: ```suggestion
The suggested content for an unused `layers` array is the [filler descriptor](#guidance-for-a-filler-descriptor).
```

}
```

Here is an example manifest for a simple artifact without content in the config, using the filler descriptor:

```json,title=Artifact%20without%20config&mediatype=application/vnd.oci.image.manifest.v1%2Bjson
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "artifactType": "application/vnd.example+type",
  "config": {
    "mediaType": "application/vnd.oci.filler.v1+json",
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "size": 2

>   },

Reviewer: It's not _super_ important, but it seems like a good idea for us to be consistent in how we order these (between the struct and all our examples).  Perhaps that's a good follow-up PR. :smile:

Author: Might as well fix it now. Done in the latest update.
